runner/logs.go

```diff
@@ -51,8 +51,36 @@ func downloadLogsForServer(server state.Server, globalCollectionOpts state.Colle
 	return newLogState, true, nil
 }
 
+func downloadLogsFromOneServer(wg *sync.WaitGroup, server *state.Server, globalCollectionOpts state.CollectionOpts, logger *util.Logger) {
+	defer wg.Done()
+	prefixedLogger := logger.WithPrefixAndRememberErrors(server.Config.SectionName)
+
+	server.CollectionStatusMutex.Lock()
+	if server.CollectionStatus.LogSnapshotDisabled {
+		server.CollectionStatusMutex.Unlock()
+		return
+	}
+	server.CollectionStatusMutex.Unlock()
+
+	server.LogStateMutex.Lock()
+	newLogState, success, err := downloadLogsForServer(server, globalCollectionOpts, prefixedLogger)
+	if err != nil {
+		server.LogStateMutex.Unlock()
+		prefixedLogger.PrintError("Could not collect logs for server: %s", err)
+		if server.Config.ErrorCallback != "" {
+			go runCompletionCallback("error", server.Config.ErrorCallback, server.Config.SectionName, "logs", err, prefixedLogger)
+		}
+	} else if success {
```

Comment:
It might be better to change the code like this, fixing a problem with the previous code I just discovered whilst reading this (mutex wasn't unlocked when `err == nil && !success`)

```golang
} else {
  server.LogPrevState = newLogState
  server.LogStateMutex.Unlock()
  if success && server.Config.SuccessCallback != "" {
    go runCompletionCallback("success", server.Config.SuccessCallback, server.Config.SectionName, "logs", nil, prefixedLogger)
  }
}
```